### PR TITLE
Add XML documentation to public properties, methods, and classes.

### LINKED ISSUE #78: Add XML Documentation Comments

### PR DESCRIPTION
**Pull request by natemcmaster**
Fix #78

### CODE REVIEW

**Comment by bricelam** [comment]
cc @rowanmiller

File: src/Microsoft.Data.Sqlite/CacheMode.cs [hunk 1]
```diff
@@ -3,10 +3,20 @@
...
     public enum CacheMode
     {
         // TODO consider adding a 'default' or 'unspecified' mode
+        /// <summary>
+        /// Default mode. Each connection uses a private cache.
+        /// </summary>
         Private,
+
+        /// <summary>
+        /// Shared-cache mode. Connection share a cache. This mode can change the behavior of transaction and table locking.
```
Comment (Eilon) [inline src/Microsoft.Data.Sqlite/CacheMode.cs]: Is this PR still alive? If so, change `Connection` to `Connections` in the second sentence.

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/CacheMode.cs]: Somewhat alive. Fixed the typo.

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs [hunk 2]
```diff
@@ -8,6 +8,9 @@
 
 namespace Microsoft.Data.Sqlite
 {
+    /// <summary>
+    /// Represents a connection with a SQLite database
```
Comment (Eilon) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: Recommend ensuring all the doc comments are full sentences (I think just missing periods in some places).

**Comment by Eilon** [comment]
@natemcmaster thanks - I just happened to be looking at some older unmerged PRs and happened to see this. :shipit: after adding some `.`s.

**Pull request merged**